### PR TITLE
Lay foundations for `pint` Quantity objects for EUI, Areas on PropertyStates

### PR DESCRIPTION
**Pull request by rgm**
#### Any background context you want to provide?

The EUIs and areas in the PropertyState are stored as simple floats, and are implicitly kBtu/ft2/year / square feet. Incoming data from Canadian sources is going to be various flavours of metric.

#### What's this PR do?

Introduces 7 new PropertyState columns for unit-tagged EUIs and areas. Lays a UI foundation—hidden behind a dark-launch flag until the QA work around working with metric units is completed—to allow a user to specify mappings and data quality rules with a selection of units. This will enable working with heterogeneous data files (eg. some columns metric, others imperial), and to understand data quality rules in a user's preferred units without having to do mental translation (eg. 15 kWh/m2/year for Passive House).

#### How should this be manually tested?

Because this is foundational, with more non-database work to come, leave the `release:use_pint` flipper off in Angular and Python and confirm that there are no user-visible changes and all tests continue to pass.

#### What are the relevant tickets?

n/a

#### Definition of Done:

- [x] Is there appropriate test coverage? (e.g. ChefSpec, Mocha/Chai, Python, etc.)
- [x] Does this PR require a Selenium test? (e.g. Browser-specific bugs or complicated UI bugs)
- [x] Does this PR require a regression test? All fixes require a regression test.
- [x] Does this add new dependencies? If so, does PIP, npm, bower requirements need to be updated?

### CODE REVIEW

**Comment by rgm** [comment]
@nllong submitting for your review prior to feature freeze on Sep 14. Remaining work for 2.2:

- [ ] fix broken tests
- [ ] figure out a snag I hit with the `PropertyState` serializer in c6b8673, 3f29578, efdf6b6 (which I'm hoping you as a more knowledgeable django rest frameworker you can help me understand).

Deferring to the follow-up PR where we turn on the pint columns:

- hook up units into mapping suggestions (ie. a raw column should prepopulate with its last known units once it's been mapped once)
- UI for selecting the front-end units to display.

File: seed/data_importer/tasks.py [hunk 1]
```diff
@@ -253,15 +286,15 @@ def map_row_chunk(ids, file_pk, source_type, prog_key, increment, **kwargs):
     # Ideally the table_mapping method would be attached to the import_file_id, someday...
     list_of_raw_columns = import_file.first_row_columns
     if list_of_raw_columns:
-        for k, v in table_mappings.items():
-            for key2 in v.keys():
-                if key2 not in list_of_raw_columns:
-                    del table_mappings[k][key2]
+        for table, mappings in table_mappings.items():
```
Comment (nllong) [inline seed/data_importer/tasks.py]: thanks for making this explicit!

File: seed/data_importer/tasks.py [hunk 2]
```diff
@@ -275,7 +308,7 @@ def map_row_chunk(ids, file_pk, source_type, prog_key, increment, **kwargs):
         table_mappings['PropertyState'] = debug_inferred_prop_state_mapping
     # TODO: *END TOTAL TERRIBLE HACK**
 
-    map_cleaner = _build_cleaner(org)
+    map_cleaner = _build_cleaner_2(org)
```
Comment (nllong) [inline seed/data_importer/tasks.py]: i assume we deprecate _build_cleaner when we move over to _build_cleaner_2? Does _build_cleaner exist anymore? If so, can you mark it as to be deprecated?

File: seed/models/columns.py [hunk 3]
```diff
@@ -259,6 +260,7 @@ def create_mappings(mappings, organization, user, import_file_id=None):
                 cache_column_mapping.append(
                     {
                         'from_field': mapping['from_field'],
+                        'from_units': mapping.get('from_units'),
```
Comment (nllong) [inline seed/models/columns.py]: any reason not to conver the other items in this dict to mapping.get('*') to make it consistent?

File: seed/models/data_quality.py [hunk 4]
```diff
@@ -235,6 +241,44 @@ class ComparisonError(Exception):
...
+    pretty_source_units = pretty_units(source_value)
+    pretty_rule_units = pretty_units(rule_value)
+
+    if incoming_data_units != rule_units:
+        formatted_value = u"{:.1f} {} → {:.1f} {}".format(
+            source_value.magnitude, pretty_source_units,
+            rule_value.magnitude, pretty_rule_units,
+        )
+    else:
+        formatted_value = u"{:.1f} {}".format(source_value, pretty_rule_units)
```
Comment (nllong) [inline seed/models/data_quality.py]: is this limiting the decimals to only the tenths place?

File: seed/data_importer/tasks.py (hunk 2 above)
Comment (rgm) [reply seed/data_importer/tasks.py]: maybe I should just replace the body of `_build_cleaner` for a clean diff, but I didn't want to do that without checking with you if that old units code was truly dead.

File: seed/models/columns.py (hunk 3 above)
Comment (rgm) [reply seed/models/columns.py]: To me, anyway, it's signalling that `from_units` is optional, which I think is the only item for which that's true. Could also give it a name if the explicit optional meaning is important:

```py
new_mapping = {'from_field': mapping['from_field'] ... }
new_mapping['from_units'] = mapping.get('from_units')  # could be missing; depends on mapping
cache_column_mapping.append(new_mapping)
```

File: seed/models/data_quality.py (hunk 4 above)
Comment (rgm) [reply seed/models/data_quality.py]: Yeah. I tried out the `{:~P}` that Pint provides and was generally getting 10-15 decimal places and it was making the data quality message hard to grok. I don't know what the right number of decimal places is.

(`{:~.1P}`, annoyingly, would format the quantity in scientific, eg `1+e04 m²`).

**Comment by nllong** [comment]
hey @rgm -- i was hoping to get this in sooner than later. Wondering if you had a chance to look at the 4 failures. 1 is just a data type issue around extra data (which isn't a big deal), the other 3 though are:

```
Traceback (most recent call last):
  File "/home/travis/build/SEED-platform/seed/seed/tests/test_views.py", line 1502, in test_get_taxlot
    self.assertEqual(properties[0]['cycle']['name'], self.cycle.name)
TypeError: 'int' object has no attribute '__getitem__'
```

**Comment by rgm** [comment]
@nllong I'm working on the test failures today. I suspect the `TypeError: 'int' object has no attribute '__getitem__'` failures are related to that serializer weirdness that I'm having some trouble figuring out.

**Comment by rgm** [comment]
@nllong yep, had some inventory serializer stuff that I should just pull out entirely and re-try in another PR. Re: the type issue, I assume I just back out ca73136a12 ... but I'm curious if you know the root cause?

**Comment by nllong** [comment]
i'm not sure at the moment how to integrate the dataType piece nor why this is happening. I just tripped across part of that code this afternoon (working on display_name). I think this is good for now. I'm going to merge tomorrow unless I hear anything from you, @axelstudios, or @Myoldmopar. Thanks!

**Comment by rgm** [comment]
Ok. Let me pull all those WIP commits and we'll restore them (in working order) next PR.

**Comment by nllong** [comment]
sounds good. let me know when it is 'ready ready'

**Pull request merged**